### DISCUSSION TITLE
Realtime, streams for queries

### DISCUSSION THREAD

**Discussion by freesailor** [supabase/supabase#3748] [Questions] [answered]
Does Supabase for dart support realtime, streams for queries or only for tables.
I need to listen to changes in a recordset returned by a .from().select() statement.

**Comment by dshukertjr** [comment, marked as answer]
Hi @freesailor!

Currently you cannot directly listen to a `select()` query, but you can listen to realtime updates of certain rows using a single `eq` filter. 

> You can listen to individual rows using the format {table}:{col}=eq.{val} - where {col} is the column name, and {val} is the value which you want to match. This syntax is the as how you can filter data in Realtime

```dart
supabase
  .from('countries:id=eq.120')
  .stream()
  .execute();
```

You can read more on how to use the `stream()` method [here](https://supabase.io/docs/reference/dart/stream).

**Reply by spydon** [reply]
Hi,
can one listen to realtime updates from a view (which then essentially can be a query)?

**Reply by GaryAustin1** [reply]
Realtime does not work on views.

**Reply by Eddydpyl** [reply]
Hi is this still the case? Does realtime not work on views?

**Reply by GaryAustin1** [reply]
It does not work on views.  It relies on a feature of Postgres that works at the table level.
